spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/CassandraConnectorConf.scala

```diff
@@ -186,8 +186,10 @@ object CassandraConnectorConf extends Logging {
   val QueryRetryParam = ConfigParameter[Int](
     name = "spark.cassandra.query.retry.count",
     section = ReferenceSection,
-    default = 10,
-    description = """Number of times to retry a timed-out query""")
+    default = -1,
```

Comment:
I think if you get the point that a request is retried 10 times, it is probably not going to succeed or the system is in bad state, so I think it is more reasonable as a default than infinite.   That said I do like that it can be configurable in some way ("Just try until it works").